spec/dummy/lib/soft_deletable_model_callbacks.rb

```diff
@@ -17,33 +21,41 @@ def self.included(base)
       around_restore :call_around_restore
       after_restore  :call_after_restore
 
+      after_commit :call_after_commit, on: :destroy
+
       def call_before_destroy
-        @before_destroy_called = true
+        @before_destroy_called = next_callback_order
       end
 
       def call_around_destroy
+        @around_destroy_called = next_callback_order
```

Comment:
Not sure why you move the called before `yield`, I think when around finishes after `after_*` callback:
https://github.com/jch/as_callbacks_tutorial/blob/master/example.rb#L71-L77